### PR TITLE
chore: upgrade to flutter 3.13.0

### PR DESCRIPTION
**Pull request by LucasXu0**
### Feature Preview

- upgrade flutter to 3.13.0
- fix some ui issues

the text-align toolbar item is too small.
before
Screenshot 2023-09-13 at 16 54 11
after
Screenshot 2023-09-13 at 17 09 21

the scroll behavior of the outline plugin is too slow.

https://github.com/AppFlowy-IO/AppFlowy/assets/11863087/01462718-769a-4da9-9257-140a6e0d8be4

add a configuration to control whether to show the rename dialog when creating a view or not.
Screenshot 2023-09-13 at 22 06 11

---

#### PR Checklist

- [ ] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/database_view/application/tar_bar_bloc.dart [hunk 1]
```diff
@@ -64,7 +64,7 @@ class GridTabBarBloc extends Bloc<GridTabBarEvent, GridTabBarState> {
             if (updatePB.createChildViews.isNotEmpty) {
               final allTabBars = [
                 ...state.tabBars,
-                ...updatePB.createChildViews.map((e) => TarBar(view: e))
+                ...updatePB.createChildViews.map((e) => TarBar(view: e)),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/application/tar_bar_bloc.dart:67]: Is `TarBar` named correctly?

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/custom_cover_picker.dart [hunk 2]
```diff
@@ -322,7 +322,7 @@ class _CoverImagePreviewWidgetState extends State<CoverImagePreviewWidget> {
                     (l) => _buildImageDeleteButton(context),
                     (r) => Container(),
                   )
-                : Container()
+                : Container(),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/custom_cover_picker.dart:325]:
```suggestion
                : const SizedBox.shrink(),
```

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/custom_cover_picker.dart [hunk 3]
```diff
@@ -322,7 +322,7 @@ class _CoverImagePreviewWidgetState extends State<CoverImagePreviewWidget> {
                     (l) => _buildImageDeleteButton(context),
                     (r) => Container(),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/custom_cover_picker.dart:323]:
```suggestion
                    (r) => const SizedBox.shrink(),
```

File: frontend/appflowy_flutter/lib/workspace/application/settings/create_file_settings_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  Future<void> getInitialSettings() async {
+    final future = getIt<KeyValueStorage>().getWithFormat(
+      KVKeys.showRenameDialogWhenCreatingNewFile,
+      (value) => bool.parse(value),
+    );
+    future.then(
+      (value) => value.fold((l) => emit(false), (r) => emit(r)),
+    );
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/application/settings/create_file_settings_cubit.dart:26]:
```suggestion
    final settingsOrFailure = await getIt<KeyValueStorage>().getWithFormat(
      KVKeys.showRenameDialogWhenCreatingNewFile,
      (value) => bool.parse(value),
    );
    
    value.fold((_) => emit(false), (settings) => emit(settings));
```

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/create_file_setting.dart [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import 'package:appflowy/workspace/application/settings/prelude.dart';
+import 'package:appflowy/workspace/presentation/settings/widgets/settings_appearance/theme_setting_entry_template.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class CreateFileSettings extends StatelessWidget {
+  CreateFileSettings({
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/create_file_setting.dart:7]:
```suggestion
  const CreateFileSettings({
```

**Review by Xazin — APPROVED** [review]

File: frontend/appflowy_flutter/lib/plugins/database_view/application/tar_bar_bloc.dart (hunk 1 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/plugins/database_view/application/tar_bar_bloc.dart:67]: No, but tt's not related to this PR. Let me fix it in the next PR.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/custom_cover_picker.dart (hunk 2 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/custom_cover_picker.dart:325]: I will add a linter check and replace all the `Container()` with `SizedBox.shrink()`;

File: frontend/appflowy_flutter/lib/workspace/application/settings/create_file_settings_cubit.dart (hunk 4 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/workspace/application/settings/create_file_settings_cubit.dart]: 👍🏻

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/create_file_setting.dart (hunk 5 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/create_file_setting.dart:7]: the const constructor can't have a body.

**Comment by LucasXu0** [comment]
half of the integration tests failed after upgrading flutter to 3.13.0. I have submitted a new PR to merge the bugs first and handle the failed integration test later.

**Pull request closed without merging**